src/actions/accountEntryActions.js

```diff
@@ -0,0 +1,74 @@
+import { trimObjValues } from "../util/utilFunctions.js";
+
+export const signup = newUserData => {
+  return new Promise((resolve, reject) => {
+    const alreadyFoundError = new Error("usernameAlreadyExists");
+    if (window.db) {
+      const newUser = trimObjValues(newUserData);
+      const db = window.db;
+      const auth = window.auth;
+      db.doc(`/users/${newUser.username}`)
+        .get()
+        .then(doc => {
+          if (doc.exists) {
+            throw alreadyFoundError;
+          } else {
+            return auth.createUserWithEmailAndPassword(
+              newUser.email,
+              newUser.password
+            );
+          }
+        })
+        .then(data => {
+          const userCredentials = {
```

Comment:
Optional improvement: Create a model / POJO for this User Credential - https://g-liu.com/blog/2015/08/object-oriented-programming-javascript-using-pojos-for-good/ (in a different file)

Generally it's better to separate out your data from your controller/actions from your UI/views. This is called the MVC model (model - view - controller): https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller